neuromation/api/utils.py

```diff
@@ -127,3 +129,54 @@ def find_project_root(path: Optional[Path] = None) -> Path:
             return here
         here = here.parent
     raise ConfigError(f"Project root is not found for {path}")
+
+
+QueuedCall = Callable[[], Any]
+
+
+def queue_calls(
+    any_obj: Any, allow_any_for_none: bool = True,
+) -> Tuple["asyncio.Queue[QueuedCall]", Any]:  # Sadly, but there is now way to annotate
+    """Add calls to asyncio.Queue instead executing them directly
+
+    Wraps given object into proxy, so trying to call any of its method will produce
+    a coroutine, that add QueuedCall to queue. For example, the following code:
+
+    class Foo:
+        def bar(self, arg):
+            print(arg)
+    queue, wrapped = queue_calls(Foo())
+    await wrapped.bar("foo")
+
+    Will add QueuedCall(foo.bar, args=("foo",)) to the queue and will not print
+    anything.
+
+    To execute calls, you can do next:
+
+    queued_call = await queue.get()
+    queued_call.execute()
+
+    In case `any_obj` is `None` and `allow_any_for_none` is set, then a proxy will not
+    raise any AttributeErrors and just absorb all cals silently.
+    """
+    queue: "asyncio.Queue[QueuedCall]" = asyncio.Queue()
+
+    async def add_to_queue(
+        real_method: Callable[..., None], *args: Any, **kwargs: Any
+    ) -> None:
+        if not callable(real_method):
+            # Force TypeError
+            real_method()  # NOQA
+        await queue.put(partial(real_method, *args, **kwargs))
+
+    async def noop(*args: Any, **kwargs: Any) -> None:
+        pass
+
+    class Proxy:
+        def __getattr__(self, item: str) -> Callable[[Any], Coroutine[Any, Any, None]]:
+            if any_obj is None and allow_any_for_none:
```

Comment:
This condition does not change its value, so you can check it at the time of creation the `Proxy` class.